### ISSUE TITLE
The module doesn't clean up after itself

### ISSUE THREAD

**Issue by alanmels** [backdrop-contrib/ip#14]
After disabling and uninstalling the module, I had to manually delete the IP demerits related views and the `/files/config_*/active/ip_demerits.settings.json` file. 

Also I had to go to pma to delete number of IP-related database tables:

<img width="1083" alt="Screenshot 2023-08-15 at 4 56 15 AM" src="https://github.com/backdrop-contrib/ip/assets/56604405/456b89ae-1bd6-47de-b444-66f25c285d94">

I believe the module needs to implement the `hook_uninstall()` to clean up all its traces.

**Comment by bugfolder** [comment]
The `ip_demerits.settings.json` file and its views are created and managed by the `ip_demerits` submodule. When you say 

> After disabling and uninstalling the module, 

are you referring to the `ip_demerits` submodule?

I've just tested this. I installed the `ip_demerits` submodule, which creates the `/files/config_*/active/ip_demerits.settings.json` file and some views, then disabled and uninstalled it, which removed the file and views associated with that module.

There's no need to to implement `hook_uninstall()` for config files that are declared in `hook_config_info()`, which is the case with the `ip_demerits` submodule. Similarly, there's no need to remove tables that were declared in `hook_schema()`.

Are you sure you actually uninstalled the submodule, not just disabled it? Disabling leaves files and views in place.
